lib/main.dart

```diff
@@ -13,51 +11,36 @@ class MyApp extends StatelessWidget {
 
   @override
   Widget build(BuildContext context) {
-    return MaterialApp(
-        title: 'Solaris App',
-        theme: ThemeData(
-          primarySwatch: Colors.blue,
-        ),
-        home: const AppScreen());
+    return MaterialApp.router(
+      routerConfig: router,
+    );
   }
 }
 
-class AppScreen extends StatefulWidget {
-  const AppScreen({super.key});
+class AppScaffold extends StatefulWidget {
+  final Widget child;
+
+  const AppScaffold({super.key, required this.child});
 
   @override
-  State<AppScreen> createState() => _AppScreenState();
+  State createState() => _AppScaffoldState();
 }
 
-class _AppScreenState extends State<AppScreen> {
-  var selectedPageIndex = 0;
-  Widget page = const HomeScreen();
-
+class _AppScaffoldState extends State<AppScaffold> {
   @override
   Widget build(BuildContext context) {
-    switch (selectedPageIndex) {
-      case 0:
-        page = const HomeScreen();
-        break;
-      case 1:
-        page = const TransferScreen();
-        break;
-      case 2:
-        page = const HubScreen();
-        break;
-    }
-
     return Scaffold(
-        appBar: AppBar(title: const Text("Solaris App")),
-        bottomNavigationBar: BottomNavigationBar(
-            currentIndex: selectedPageIndex,
-            onTap: (pageIndex) => setState(() => selectedPageIndex = pageIndex),
-            items: const [
-              BottomNavigationBarItem(icon: Icon(Icons.home), label: "Home"),
-              BottomNavigationBarItem(
-                  icon: Icon(Icons.search), label: "Transfer"),
-              BottomNavigationBarItem(icon: Icon(Icons.person), label: "Hub")
-            ]),
-        body: page);
+      appBar: AppBar(title: const Text('Solaris')),
+      body: widget.child,
+      bottomNavigationBar: BottomNavigationBar(
+        currentIndex: calculateSelectedIndex(context),
+        onTap: (value) => onTap(value, context),
+        items: const [
```

Comment:
What do you think about keeping these 3 options in a data structure (ex: an array or hashmap) and use the same data structure here (for `items`) and on `currentIndex` and on `onTap` ?
In this way I believe we won't even need `calculateSelectedIndex` and `onTap` anymore (or they would be way simpler), since we would use the index and route value from the data structure.
@ilie-lupu-ts @szaszgyozo